src/System.Linq.Queryable/src/System/Linq/EnumerableRewriter.cs

```diff
@@ -176,7 +174,7 @@ private Type GetEquivalentType(Type type)
                     var interfacesWithInfo = info.ImplementedInterfaces.Select(i => new { Type = i, Info = i.GetTypeInfo() }).ToArray();
                     var singleTypeGenInterfacesWithGetType = interfacesWithInfo
                         .Where(i => i.Info.IsGenericType && i.Info.GenericTypeArguments.Length == 1)
-                        .Select(i => new { Type = i.Type, Info = i.Info, GenType = i.Info.GetGenericTypeDefinition() });
+                        .Select(i => new { i.Type, i.Info, GenType = i.Info.GetGenericTypeDefinition() }).ToList();
```

Comment:
We don't even need it back afterwards, so there's no downside in cutting it, and we lose an anonymous type and a delegate (replace with method group) above too. I'm not sure what that was doing there. I think it just seemed likely to be useful when I was thinking it out, and it survived the subsequent tidying.